### PR TITLE
Spiralogram: Upgrade color palette to use the new colorpalettes module

### PR DESCRIPTION
**Pull request by ajdapretnar**
##### Issue

Color palettes have changed in core Orange.

##### Description of changes
Update to latest palettes.

To discuss: which palette to use. I think it was nicer before - BlGrYl is not the best for this. Comments welcome.

##### Includes
- [X] Code changes
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

**Comment by PrimozGodec** [comment]
For me, the previous palette was more intuitive when I saw it first. The current one needs a bit more time to understand what is high/low. At least I am missing some legend (what different color means).

**Comment by ajdapretnar** [comment]
Yes, I agree. We can implement custom palette for this widget. Should we go with the old color (red) or try something new?

As for the legend... Be my guest. 😆 This is highcharts magic.

**Comment by janezd** [comment]
I see no problem with widget-specific palettes. I would just recommend using the same classes, just with specific colors. From colorcet, if possible. But I prefer the old colors (red), too.

**Comment by kernc** [comment]
Of colorcet, I only find [blues](https://colorcet.holoviz.org/#Samples) intuitively linear. It's certainly got something to do with the number of distinct colors in the palette and the background.

**Comment by ajdapretnar** [comment]
Fixed. Used linear_blue_5-95_c73 from colorcet. I think it is... okay-ish. Could be better.

**Comment by janezd** [comment]
We prefer red. Consider:

- migrate existing colors
- swap red and blue in colorcet --- is it still linear then?
- use fire or inferno

**Comment by ajdapretnar** [comment]
Options:

1. Old Spiralogram with selection:
Screen Shot 2020-04-03 at 11 52 11

2. New Spiralogram with linear blue and selection:
Screen Shot 2020-04-03 at 11 53 04

3. Swapping red and blue in colorcet with selection:
Screen Shot 2020-04-03 at 12 02 37

4. Inferno with selection:
Screen Shot 2020-04-03 at 12 04 10

**Comment by ajdapretnar** [comment]
I still think the old options is the best. I have not yet managed to migrate it to Spiralogram, will play around.

**Comment by ajdapretnar** [comment]
Below is the old palette in the new version.
Problem: span of the old palette is not 256.
I filled the missing 51 values by inserting R,G+1,B to every fourth position in the list, but then the palette is not really linear. Ideas? 

Screen Shot 2020-04-03 at 13 31 16

**Comment by janezd** [comment]
Inserting at every fourth place keeps it linear (within rounding error), if it was linear before (which it probably wasn't). It the aim of this PR is porting, now improving, I'm OK with it.

**Comment by ajdapretnar** [comment]
Ok, I push.

**Comment by kernc** [comment]
Since it's a one color palette, can we maybe keep stronger pigment indicating higher values?

> Problem: span of the old palette is not 256. I filled the missing 51 values ...

I fail to fully grasp the issue here, but isn't that something to be fixed on `ContinuousPalette` (note, _continuous_)?

File: orangecontrib/timeseries/widgets/owspiralogram.py [hunk 1]
```diff
@@ -135,8 +140,8 @@ def setSeries(self, timeseries, attr, xdim, ydim, fagg):
             self.clear()
             return
         ptpval = maxval - minval
-        color = GradientPaletteGenerator('#ffcccc', '#cc0000')
-        selected_color = GradientPaletteGenerator('#cdd1ff', '#0715cd')
+        color = palette
+        selected_color = palette
```
Comment (kernc) [inline orangecontrib/timeseries/widgets/owspiralogram.py:144]: Why, these two palettes are equal? :confused:

**Comment by janezd** [comment]
Continuous palettes now suppose that they have a table of 256 colors, for sake of simplicity (and because all built-in palettes are based on colorcet, which have 256 colors).

We can add an interpolating constructor; I doubt that anybody would ever want to define a palette with more than 256 colors. But I'm also not sure how generally useful will it be.

**Comment by ajdapretnar** [comment]
As for the selection, I am debating whether we should use a single color or a palette? I think a single color is sufficient, no?

**Comment by ajdapretnar** [comment]
@janezd This is now kind of fixed. I imitated the old red color and set the selection for continuous data to our discrete blue. I kind of like it, it is very explicit.

**Comment by janezd** [comment]
I don't like the single color for selection. I think it hides information for no good reason. This is just a vote.

1. If we have a single color, don't construct `LimitedDiscretePalette(1)` and then take the first color. Just construct a `QColor`.

    In this case, also note that blue color is used only for numeric variables, while discrete variables (above) use darker shades. I supposed this would have to be unified then.

2. I would prefer to keep colored selections, though. I think they are visible enough.

    Screenshot 2020-04-07 at 18 30 46

    In this case, the code can be refactored so that there is no separate `sel_color`.

**Comment by janezd** [comment]
> I imitated the old red color 

What does "imitating the old red color" mean? In particular, is this palette linear? I don't object if it's not, just rename it from `linear_red` to something like `red_gradient`.

**Comment by ajdapretnar** [comment]
I agree, 2. looks better.
Imitating the old red color means creating linear scale going from #ffcccc to #cc0000, then inserting a slightly different shade at every 4th place to expand it to 256. I don't know if this still counts as linear, it would be like this: (204, 0, 0), **(204, 1, 0)**, (204, 1, 1), (204, 2, 2), (204, 3, 3). The bold is the inserted one.
I can rename it to whatever.

**Comment by janezd** [comment]
> Imitating the old red color means creating linear scale 

I see. :) "Linear" palettes are supposed to be perceptually linear, not linear in RGB space (which is not aligned well with our cognitive processing). So I suppose it should be renamed to whatever, just not literally. :)

**Comment by ajdapretnar** [comment]
Name changed.

**Comment by ajdapretnar** [comment]
Btw, selection is not really all that nice though, since not the entire field is encircled with black. Inner line is always missing. Does it bother anyone else beside me?

**Pull request merged**